### ISSUE TITLE
Still getting JSON error after all fixes

### ISSUE THREAD

**Issue by tabletguy** [robwent/joomla-json-db-check#11]
From the article https://www.joomlashack.com/blog/tutorials/error-decoding-json-data-syntax/ I found this script.

However, after running the script and fixing all errors, site still gives JSON error. I modified the Json.php library per https://forum.joomla.org/viewtopic.php?t=937036 and found that there are still param fields in error. Specifically (with that library mod) I get the error " Error decoding JSON data: Syntax error the bad data is: {""}"

I'm not knowledgeable enough to know what module, etc. is generating this.

I thought I'd let you know, at least, that the script is apparently not finding this "error". Running Joomla 3.7.0

**Comment by robwent** [comment]
Did you click on the 'Check For All Invalid Values' button?.

The script will not automatically fix any errors other than empty brackets `{}`

It might be that you have an extension installed that is storing bad data in a database column that is not being checked.

**Comment by tabletguy** [comment]
Yes, I also checked for all invalid values. Fixed those and rechecked.

I've also (just now) manually viewed all tables looking for something.
Didn't find anything. I've temporarily resorted to commenting out the error
check mentioned in the Joomla forum post that I linked, so that others can
use the site (administrator) while I continue to look.

I don't know enough to perhaps expand the "throw" message to include which
table, column and row is being referenced. (I'm referring to the line about
halfway down the discussion "throw new \RuntimeException(sprintf('Error
decoding JSON data: %s the bad data is: %s', json_last_error_msg(),
$data));"

**Comment by tabletguy** [comment]
Does your script check for the {""} invalid value?  I just reread your comment where you say the script only fixed empty brackets {}?

**Comment by robwent** [comment]
Yes sorry, I meant `{""}`

Where are you actually seeing the error?

If it's the frontend of the site, then disable all modules and plugins and check that the error is gone and then turn them back on again until the error reappears.

If it's in the admin area then you should have a clue of where the problem comes from by the page where the message appears.

**Comment by tabletguy** [comment]
I've reenabled the error check code and logged out and back in a couple of time and am not getting the error now.

I had cleared all cache files and made sure the site wasn't being cached (Cloudflare in this case), etc.

Even though I thought I had cleared all browser cache files as well, that's the only explanation to why it started working, although it took a couple of hours to get it working.

I hadn't tried using a different browser right away (which would have shown me if fixes worked even if I had browser pages cached).

So, at this point, it's looking like it's all working.
